### PR TITLE
Initial onboarding overhaul

### LINKED ISSUE #806: Onboarding: Remove `installation-progress` and install plugin in the background
The `installation-progress` step brings no real value to users and can be instead confusing. Let's remove this step and instead, when the user clicks on "Get Started" for the first time, we install the theme/plugins automatically in the background.

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

First step of a larger onboarding overhaul. 
Closes #806 

Also, this PR proposes using [Tachyons CSS framework](http://tachyons.io/) for easier UI building. Note that this framework is not meant for building components or being used _in_ `newspack-components`, since it's not shipped with the versions distributed on npm. This framework is meant just as a lightweight tool for faster UI building.

### How to test the changes in this Pull Request:

1. Install the plugin on a fresh WP site 
2. Observe a much simplified onboarding experience:
3. All required software (plugins, themes), as well as starter content, is installed in the background, with only a progress bar as UI feedback
4. If any step of this fails, a warning will be displayed. User has to reload the page to resume.
5. The profile information is stripped to just country & currency
6. Jetpack, Site Kit, and Mailchimp connections are all on one view
7. After choosing the theme the setup is complete. 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: assets/wizards/setup/views/welcome/index.js [hunk 1]
```diff
@@ -1,50 +1,198 @@
...
+					.catch( addError( { info: ERROR_TYPES.plugin_configuration, item: item.Name } ) );
+		} );
+		for ( let i = 0; i < softwarePromises.length; i++ ) {
+			await softwarePromises[ i ]();
+		}
+
+		// Starter content.
+		await starterContentFetch( `categories` )
+			.then( increment )
+			.catch( addError( { info: ERROR_TYPES.starter_content, item: 'categories' } ) );
```
Comment (dkoo) [inline assets/wizards/setup/views/welcome/index.js:91]: The error messages for if these fail are pretty blunt—could we add some short description, e.g. "Failed to create starter content: homepage" or "Failed to install plugin: [PLUGIN NAME]"?

Screen Shot 2021-02-05 at 11 59 56 AM

**Review by dkoo — CHANGES_REQUESTED** [review]
Love this simplified onboarding process. I have a few concerns/requests, mostly in the installation step. Let me know if anything is unclear or if it runs counter to an intentional design decision!

File: assets/wizards/setup/views/welcome/index.js (hunk 1 above)
Comment (adekbadek) [reply assets/wizards/setup/views/welcome/index.js]: Done in 8267401ad3cc851fed4a29794a5b79a36e75c1a2

File: assets/wizards/setup/views/welcome/index.js [hunk 2]
```diff
@@ -1,50 +1,198 @@
...
+				'There has been an error during the installation. Please retry or manually install required plugins to continue with the configuration of your Newspack site.'
+			);
+		}
+		if ( isInit ) {
+			return __(
+				'We’ll help you get set up by installing the most relevant Newspack plugins in the background.'
+			);
+		}
+		if ( isDone ) {
+			return __( 'Click the button below to proceed.' );
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply assets/wizards/setup/views/welcome/index.js:161]: I've added it to a ["later improvements" issue.](https://github.com/Automattic/newspack-plugin/issues/852)

File: assets/wizards/setup/views/welcome/index.js [hunk 3]
```diff
@@ -1,50 +1,198 @@
...
+		if ( isDone ) {
+			return __( 'Installation complete!' );
+		}
+		return __( 'Installing…' );
+	};
+
+	const getInfoText = () => {
+		if ( hasErrors ) {
+			return __(
+				'There has been an error during the installation. Please retry or manually install required plugins to continue with the configuration of your Newspack site.'
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply assets/wizards/setup/views/welcome/index.js:152]: > is it possible to provide a "Retry" button

I've added it to a ["later improvements" issue.](https://github.com/Automattic/newspack-plugin/issues/852)

> any starter content that had been successfully created on previous attempts will be created again

Fixed in bd30e07bbc189068c96fed88695bf714e8e80446

File: assets/wizards/setup/views/welcome/index.js [hunk 4]
```diff
@@ -1,50 +1,198 @@
...
+		await starterContentFetch( `theme` )
+			.then( increment )
+			.catch( addError( { info: ERROR_TYPES.starter_content, item: 'theme' } ) );
+	};
+
+	const hasErrors = errors.length > 0;
+	const isInit = installationProgress === 0;
+	const isDone = installationProgress === TOTAL;
+
+	// eslint-disable-next-line no-nested-ternary
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply assets/wizards/setup/views/welcome/index.js]: Removed in f521886229770a6838253b5e8d968b419051208f

File: assets/wizards/setup/views/settings/style.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+/**
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply assets/wizards/setup/views/settings/style.scss]: Whoops, removed in f521886229770a6838253b5e8d968b419051208f

File: assets/wizards/setup/views/settings/index.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+			} )
+			.catch( setError );
+	}, [] );
+
+	return (
+		<Fragment>
+			<SelectControl
+				disabled={ isLoading }
+				label={ __( 'Where is your business based?' ) }
+				value={ profileData.country }
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply assets/wizards/setup/views/settings/index.js:40]: Good idea, done in 05abedc60770d927953654352b0cc7044e6107a3

File: assets/wizards/setup/views/integrations/index.js [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+		);
+	}
+	if ( integration.url ) {
+		return (
+			<Button isLink href={ integration.url } target="_blank">
+				{ integration.buttonText }
+			</Button>
+		);
+	}
+	if ( integration.error?.code === 'unavailable_site_id' ) {
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply assets/wizards/setup/views/integrations/index.js:93]: That's true, but it's an issue with either WPCOM or Jetpack. I've submitted an issue [here](https://github.com/Automattic/jetpack/issues/18747). Fixed in 536229688da55da497fd48cbad4c1cb67b8feb30, also for the engagement wizard which suffered from the same issue.

**Review by dkoo — APPROVED** [review]
Thanks for opening the improvements issue—the other fixes look good to me. 👍

**Comment by adekbadek** [comment]
@thomasguillot Can you take a look?

**Review by thomasguillot — APPROVED** [review]

**Comment by thomasguillot** [comment]
It's a great first step, thanks @adekbadek 😊

**Pull request merged**